src/Orleans.Core/Messaging/GatewayConnection.cs

```diff
@@ -250,28 +250,48 @@ protected override void OnGetSendingSocketFailure(Message msg, string error)
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
         protected override void OnMessageSerializationFailure(Message msg, Exception exc)
         {
-            // we only get here if we failed to serialise the msg (or any other catastrophic failure).
-            // Request msg fails to serialise on the sending silo, so we just enqueue a rejection msg.
-            Log.Warn(ErrorCode.ProxyClient_SerializationError, $"Unexpected error serializing message to gateway {Address}.", exc);
-            FailMessage(msg, $"Unexpected error serializing message to gateway {Address}. {exc}");
-            if (msg.Direction == Message.Directions.Request || msg.Direction == Message.Directions.OneWay)
+            // we only get here if we failed to serialize the msg (or any other catastrophic failure).
+            // Request msg fails to serialize on the sender, so we just enqueue a rejection msg.
+            // Response msg fails to serialize on the responding silo, so we try to send an error response back.
+            if (this.Log.IsEnabled(LogLevel.Warning))
```

Comment:
nit: I don't think we need to check everywhere if logger has Warning turned on. Most people set logging equal or above info. this check turns true in most cases. 
we usually check it for Debug and Trace cases